build-logic/code-quality/src/main/kotlin/calcite.testing.gradle.kts

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import org.gradle.api.tasks.testing.Test
+import org.gradle.api.tasks.testing.logging.TestExceptionFormat
+
+plugins {
+    `java-library`
+}
+
+dependencies {
+    // TODO: add junit5 by default
+}
+
+tasks.withType<Test>().configureEach {
```

Comment:
does it mean this occurs before in every subproject and now centralize here?

Reply:
The file is named `calcite.testing.gradle.kts`.
This code will be activated only for those `.gradle.kts` files that include

```kotlin
plugins {
    id("calcite.testing") // <-- gradle would apply code from calcite.testing.gradle.kts
}
```

It allows composition of the build logic, so the individual submodules would mention the bits they need in `plugins` section.

Reply:
In other words, build files would never "apply automatically", and build logic would always be pulled by `plugins { .. }` section